src/library/features/crates/cratefeature.cpp

```diff
@@ -413,8 +461,16 @@ void CrateFeature::slotDeleteCrate() {
             qWarning() << "Refusing to delete locked crate" << crate;
             return;
         }
+        // better deletion requeried
+        if (m_crateHierarchy.hasChildern(crate.getId())) {
+            qWarning() << "Can't delete " << crate;
+            qWarning() << "Currently only delete leaf crates";
+            return;
+        }
         if (m_pTrackCollection->deleteCrate(crate.getId())) {
+            m_crateHierarchy.deleteCrate(crate.getId());
```

Comment:
The GUI code shouldn't have to be concerned with this. Calling TrackCollection::deleteCrate should take care of it automatically.